cheroot/__main__.py

```diff
@@ -9,54 +9,76 @@
     # Start a server on 0.0.0.0:9000 with 8 threads
     # for the WSGI app myapp/wsgi.py:main_app()
     cheroot myapp.wsgi:main_app --bind 0.0.0.0:9000 --threads 8
+
+    # Start a server on the UNIX socket /var/spool/myapp.sock
+    cheroot myapp.wsgi --bind /var/spool/myapp.sock
+
+    # Start a server on the abstract UNIX socket CherootServer
+    cheroot myapp.wsgi --bind @CherootServer
 """
 
 import argparse
 from importlib import import_module
 import os
-import re
 import sys
 
-from .wsgi import Server
+import six
 
-RE_HOST_PORT = r'^(\S+)(?::)([0-9]+)$'
+from .wsgi import Server
 
 
-def wsgi_app_path(string):
-    """Convert WSGI app string to a tuple."""
-    if ':' in string:
-        mod_path, app_path = string.split(':', 1)
+def get_wsgi_app(wsgi_app_path):
+    """Read WSGI app path string and import application module."""
+    if ':' in wsgi_app_path:
+        mod_path, app_path = wsgi_app_path.split(':', 1)
+        if ':' in app_path:
+            raise ValueError(
+                'Application object name is invalid: it should not contain '
+                'colon (:) character.'
+            )
     else:
-        mod_path, app_path = string, 'application'
-
-    module = import_module(mod_path)
-    app = eval(app_path, vars(module))
-
-    if app is None:
-        raise NameError(
-            'Failed to find application object: {}'.format(app_path))
-    elif not callable(app):
-        raise TypeError('Application must be a callable object')
-
-    return app
-
+        mod_path, app_path = wsgi_app_path, 'application'
 
-def wsgi_bind_addr(string):
-    """Convert bind address string to a format accepted by Server()."""
+    try:
+        wsgi_app = getattr(import_module(mod_path), app_path)
+        if not callable(wsgi_app):
+            raise TypeError
+    except ImportError as imp_err:
+        six.raise_from(NameError('Failed to find a module: {}'.format(
+            mod_path)), imp_err)
+    except AttributeError as attr_err:
+        six.raise_from(NameError(
+            'Failed to find application object: {}'.format(app_path)), attr_err
+        )
+    except TypeError as type_err:
+        six.raise_from(TypeError(
+            'Application must be a callable object'), type_err
+        )
+
+    return wsgi_app
+
+
+def parse_wsgi_bind_addr(bind_addr_string):
+    """Convert bind address string to a tuple."""
     # try and match for an IP/hostname and port
-    match = re.match(RE_HOST_PORT, string)
-    if match:
-        addr, port = match.groups()
-        port = int(port)
-        return addr, port
+    match = six.moves.urllib.parse.urlparse('//{}'.format(bind_addr_string))
+    try:
+        addr = match.hostname
+        port = match.port
+        if (addr is not None) and (port is not None):
```

Comment:
Don't abuse parens

Reply:
_[Beautiful is better than ugly.](https://zen-of-python.info/beautiful-is-better-than-ugly.html#1)_